SConstruct

```diff
@@ -363,7 +363,8 @@ MINIMAL_ENV = ('HOME', 'TERM', 'SSH_AUTH_SOCK', 'http_proxy', 'https_proxy', 'PK
 
 # Environment variables that are also kept when LD_PRELOAD is set.
 PRELOAD_ENV = ('LD_PRELOAD', 'D_LOG_FILE', 'DAOS_AGENT_DRPC_DIR', 'D_LOG_MASK', 'DD_MASK',
-               'DD_SUBSYS', 'D_IL_MAX_EQ', 'D_IL_ENFORCE_EXEC_ENV', 'D_IL_COMPATIBLE')
+               'DD_SUBSYS', 'D_IL_MAX_EQ', 'D_IL_ENFORCE_EXEC_ENV', 'D_IL_COMPATIBLE',
+               'D_IL_NO_DCACHE_BASH')
```

Comment:
im going to object on adding more env variables like this.. it's counter productive for users and hard to keep track of. if we know this has issues, then why we need a flag to enable it?

Reply:
how about if compatible mode is on, dcache in bash is disabled.
otherwise it's enabled.

Reply:
Another option would be to have a `D_IL_NO_CACHE_EXE` and have it be a comma separated list of process named with the default being `sh,bash`, that way the default would work and not need a hack in this file and there would be the option to enable/disable it for other procs as needed.

This shouldn't hold up the PR from landing however, the previous code is good enough to pass the test and land the PR IMO and we should improve this in a follow up.

Reply:
ok. I can add an env "D_IL_NO_CACHE_EXE" to support user supplied app list in a follow up PR.
I think we should disable dcahe in bash in regular mode too. bash is not sensitive to performance. Disabling cache can avoid possible consistency issue.